File: inspectit.net/InspectIT.NET Agent/StackTraceData.cpp
#include "StackTraceData.h"

#include <chrono>


StackTraceData::StackTraceData(std::vector<std::shared_ptr<StackTraceSample>> traces, JAVA_LONG platformId, JAVA_LONG methodSensorId, JAVA_LONG methodId, ThreadID threadId, JAVA_LONG timestamp)
	: MethodSensorData(platformId, methodSensorId, methodId, timestamp)
{
	this->threadId = threadId;
	this->traces = traces;
}


StackTraceData::~StackTraceData()
{
}

ThreadID StackTraceData::getThreadId()
{
	return threadId;
}

JAVA_LONG StackTraceData::getBaseMethodId() {
	return baseMethodId;
}

void StackTraceData::setBaseMethodId(JAVA_LONG baseMethodId)
{
	this->baseMethodId = baseMethodId;
}

JAVA_LONG StackTraceData::getStartTime()
{
	return startTime;
}

void StackTraceData::setStartTime(JAVA_LONG startTime)
{
	this->startTime = startTime;
}

JAVA_LONG StackTraceData::getEndTime()
{
	return endTime;
}

void StackTraceData::setEndTime(JAVA_LONG endTime)
{
	this->endTime = endTime;
}

LPWSTR StackTraceData::getJavaTypeName()
{
	return L"StackTraceData";
}

std::vector<std::shared_ptr<StackTraceSample>> StackTraceData::getTraces()
{
	return traces;
}

web::json::value StackTraceData::toJson()
{
	web::json::value json = super::toJson();
	json[L"threadId"] = web::json::value::number(threadId);
	json[L"baseMethodId"] = web::json::value::number(baseMethodId);
	json[L"startTime"] = web::json::value::number(startTime);
	json[L"endTime"] = web::json::value::number(endTime);
	
	web::json::value tracesJson;
	int i = 0;
	for (auto it = traces.begin(); it != traces.end(); it++) {
		tracesJson[i] = (*it)->toJson();
		i++;
	}

	json[L"traces"] = tracesJson;

	return json;
}

File: inspectit.shared.all/src/main/java/rocks/inspectit/shared/all/communication/data/StackTraceData.java
/**
 *
 */
package rocks.inspectit.shared.all.communication.data;

import rocks.inspectit.shared.all.communication.MethodSensorData;

/**
 * @author Henning Schulz
 *
 */
public class StackTraceData extends MethodSensorData {

	/**
	 * Generated serial UID.
	 */
	private static final long serialVersionUID = -2288816684899047450L;

	private long threadId;

	private long baseMethodId;
	private long startTime;
	private long endTime;

	private StackTraceSample[] traces;

	/**
	 * Gets {@link #threadId}.
	 *
	 * @return {@link #threadId}
	 */
	public long getThreadId() {
		return threadId;
	}

	/**
	 * Sets {@link #threadId}.
	 *
	 * @param threadId
	 *            New value for {@link #threadId}
	 */
	public void setThreadId(long threadId) {
		this.threadId = threadId;
	}

	/**
	 * Gets {@link #baseMethodId}.
	 *
	 * @return {@link #baseMethodId}
	 */
	public long getBaseMethodId() {
		return baseMethodId;
	}

	/**
	 * Sets {@link #baseMethodId}.
	 *
	 * @param baseMethodId
	 *            New value for {@link #baseMethodId}
	 */
	public void setBaseMethodId(long baseMethodId) {
		this.baseMethodId = baseMethodId;
	}

	/**
	 * Gets {@link #startTime}.
	 *
	 * @return {@link #startTime}
	 */
	public long getStartTime() {
		return startTime;
	}

	/**
	 * Sets {@link #startTime}.
	 *
	 * @param startTime
	 *            New value for {@link #startTime}
	 */
	public void setStartTime(long startTime) {
		this.startTime = startTime;
	}

	/**
	 * Gets {@link #endTime}.
	 *
	 * @return {@link #endTime}
	 */
	public long getEndTime() {
		return endTime;
	}

	/**
	 * Sets {@link #endTime}.
	 *
	 * @param endTime
	 *            New value for {@link #endTime}
	 */
	public void setEndTime(long endTime) {
		this.endTime = endTime;
	}

	/**
	 * Gets {@link #traces}.
	 *
	 * @return {@link #traces}
	 */
	public StackTraceSample[] getTraces() {
		return traces;
	}

	/**
	 * Sets {@link #traces}.
	 *
	 * @param traces
	 *            New value for {@link #traces}
	 */
	public void setTraces(StackTraceSample[] traces) {
		this.traces = traces;
	}

}
